### ISSUE TITLE
Identify when --always-make never ends

### ISSUE THREAD

**Issue by andreeis** [microsoft/vscode-makefile-tools#98]
Most of the repositories, if they have an issue with --always-make it's just that a dry-run would take a very long time and the code base would get reconfigured (probably losing any previous settings that were defined during the last ./configure).
Fewer repositories would have the issue of waiting for input from the user. We detect this with a timeout check and the user sees a bottom right message popup.
For https://github.com/swig/swig.git, when the extension configures with --always-make, something will fail (which doesn't fail when run manually from terminal) but the make build will try the same configure target again, indefinitely. This will not result in an output that never gets written into (like when the build waits for input) and which could be detected with a timeout, this results into output being indefinitely generated and we need to implement a different timeout so that the configure would force an end in this case too.

**Comment by gcampbell-msft** [comment]
Is this still something we need? @andreeis

**Comment by EchterAgo** [comment]
https://github.com/networkupstools/nut/ also loops a very long time (infinite?) with `--always-make`

**Comment by andreeis** [comment]
> Is this still something we need? @andreeis

Yes, even if PR https://github.com/microsoft/vscode-makefile-tools/pull/500 goes in, for the scenarios that are not fixed yet by that, it would be nice if we would detect when this pattern occurs and popup a timeout. It wouldn't be a hang (nothing appears to happen) but an infinite loop (same repetitive stuff happens over and over).
